node/src/main.rs

```diff
@@ -5,30 +5,28 @@ mod cli;
 mod log;
 
 use self::cli::cli;
-use forest_libp2p::service::NetworkEvent;
-use futures::prelude::*;
-use network::service::NetworkService;
+use async_std::task;
+use forest_libp2p::service::Libp2pService;
 use slog::info;
-use tokio::runtime::Runtime;
-use tokio::sync::mpsc;
+use std::error::Error;
 
-fn main() {
+#[async_std::main]
+async fn main() -> Result<(), Box<dyn Error>> {
     let log = log::setup_logging();
     info!(log, "Starting Forest");
 
     // Capture CLI inputs
     let config = cli(&log).expect("CLI error");
 
-    // Create the tokio runtime
-    let rt = Runtime::new().unwrap();
+    let logger = log.clone();
 
-    // Create the channel so we can receive messages from NetworkService
-    let (tx, _rx) = mpsc::unbounded_channel::<NetworkEvent>();
-    // Create the default libp2p config
-    // Start the NetworkService. Returns net_tx so  you can pass messages in.
-    let (_network_service, _net_tx, _exit_tx) =
-        NetworkService::new(&config.network, &log, tx, &rt.executor());
+    let lp2p_service = Libp2pService::new(logger, &config.network);
+
+    task::block_on(async move {
+        let _ = lp2p_service.run().await;
```

Comment:
```suggestion
        lp2p_service.run().await;
```
Why do you need the `let _ =` ?

Reply:
I needed it because run returns a result, but i just removed it.

Reply:
But you weren't handling the result? You needed to do something with the result or pass the error up to the main function return with `?`